src/lib/optimizer/chunk_statistics/range_filter.hpp

```diff
@@ -50,6 +70,15 @@ class RangeFilter : public AbstractFilter {
 
 template <typename T>
 std::unique_ptr<RangeFilter<T>> RangeFilter<T>::build_filter(const pmr_vector<T>& dictionary) {
```

Comment:
Could we have an optional parameter max_ranges_count that defaults to 10?

We surely have to put more logic into this number at a later point in time.